components/heap/multi_heap_internal.h

```diff
@@ -45,3 +45,41 @@ void multi_heap_internal_poison_fill_region(void *start, size_t size, bool is_fr
 void multi_heap_internal_lock(multi_heap_handle_t heap);
 
 void multi_heap_internal_unlock(multi_heap_handle_t heap);
+
+struct heap_block;
```

Comment:
Can you please give some examples the internal debugging code you've written to use this? Even though this header as marked as "internal" and not added to the public include path list, I'm a little nervous about exposing the heap implementation internals.

Maybe there are ways we can expand the existing multi_heap APIs to expose useful debug operations in a more practical way? Perhaps some sort of get_alloc_info(ptr, &info) which will return info about the heap block containing 'ptr'? Ideas (based on your own debug usage) much appreciated. :)